.github/workflows/ci.yml

```
@@ -164,6 +164,69 @@ jobs:
     - name: Setuptools helpers test
       run: pytest tests/extra_setuptools
 
+  valgrind:
+    runs-on: ubuntu-latest
+    name: "🐍 3.9 (debug) • Valgrind"
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install debug Python
+      uses: deadsnakes/action@v2.1.1
+      with:
+        python-version: 3.9
+        debug: true
+
+    - name: Update CMake
+      uses: jwlawson/actions-setup-cmake@v1.6
+
+    - name: Valgrind cache
+      uses: actions/cache@v2
+      id: cache-valgrind
+      with:
+        path: valgrind
+        key: 3.16.1 # Valgrind version
+
+    - name: Compile and install Valgrind
+      if: steps.cache-valgrind.outputs.cache-hit != 'true'
+      run: |
+        VALGRIND_VERSION=3.16.1
+        curl https://sourceware.org/pub/valgrind/valgrind-$VALGRIND_VERSION.tar.bz2 -o - | tar xj
+        mv valgrind-$VALGRIND_VERSION valgrind
+        pushd valgrind
+        ./configure && make -j 2 > /dev/null && sudo make install
+        popd
```

```suggestion
        (cd valgrind && ./configure && make -j 2 > /dev/null && sudo make install)
```
Would this be more succinct than pushd/popd?

---

Also, technically, I think you could just cd then not go back, since you don't the directory in the next line